composer/trainer/trainer.py

```diff
@@ -3310,7 +3311,8 @@ def predict_batch_end(self, state: State, logger: Logger) -> None:
                 self.state.batch = self.state.device.batch_to_device(self.state.batch)
 
                 # Perform any device transforms
-                self.state.batch = data_spec.device_transforms(self.state.batch)
+                self.state.batch = data_spec.batch_transforms(self.state.batch)
```

Comment:
shouldn't the batch transform be before the device move for consistency with other parts of the code?